### PR TITLE
Add XGBoost Feature Interactions demo notebook

### PR DESCRIPTION
**Pull request by maurever**
Prepare demo to demonstrate new xgboost features for fix release blog post.

### CODE REVIEW

File: h2o-py/demos/xgboost_feature_interactions_demo.ipynb [hunk 1]
```diff
@@ -0,0 +1,1177 @@
...
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Tree index:0\n",
+      "Node ID 0 has left child node with index 1 and right child node with index 2. The split feature is CAPSULE.\n",
+      "Node ID 1 has left child node with index -1 and right child node with index -1. The split feature is None.\n",
```
Comment (Far0n) [inline h2o-py/demos/xgboost_feature_interactions_demo.ipynb:223]: Wouldn't it be better to call them "Leafs"?

Comment (maurever) [reply h2o-py/demos/xgboost_feature_interactions_demo.ipynb:223]: I think it is not important for this example, but it is not hard to change it.

Comment (Far0n) [reply h2o-py/demos/xgboost_feature_interactions_demo.ipynb:223]: I see

**Comment by Far0n** [comment]
@maurever any specific thing you want me to look at?

**Comment by maurever** [comment]
> @maurever any specific thing you want me to look at?

Thank you @Far0n. This demo just shows how to use these new functionalities with a random dataset. If you have some suggestions on how to improve our examples to be more useful for data scientists, let us know. For example, if you have some specific task how to use these features to show benefits or whatever.

File: h2o-py/demos/xgboost_feature_interactions_demo.ipynb (hunk 1 above)
Comment (maurever) [reply h2o-py/demos/xgboost_feature_interactions_demo.ipynb]: I improved the print to show leaf nodes better.

**Comment by Far0n** [comment]
@maurever There is a little issue with `plot_tree` on my sys (tested in firefox and chrome with py3.6 and jupyter  1.0.0):

File: h2o-py/demos/xgboost_feature_interactions_demo.ipynb [hunk 2]
```diff
@@ -0,0 +1,1126 @@
...
...A2P/jgg8POrFmzwtmmTZuy+bBhw8JO9FjnzZsXdmbPnp3N16xZE3Yit956azg75JBDsvmyZcvCzm9/+9ts/tprr4Wd1atXZ/POnTuHnXXr1oWzyIYNG8rulPLd7363rJz/du6554az3r17Z/Pf/OY3Yadnz57Z/J///GfY+fGPf5zNJ02aFHZK/T1GfvSjH2Xziy++OOz8+te/DmcHHHBA2Wuoj4ULF2bzXXbZJez06tWroZbTJH7wgx+Es7lz52bzX/3qV2Fnjz322OY1/Y8iXUOU1lyv1ehaufbaa8POaaedls1vvPHGsn9/XV1dOIs+k3ziE58IO/3798/mjXWtphRfr9G1mlJ8vVbDtRq9BkevvynFf1eVfP0FPrjo8+5+++0Xdvr27dtQy6HA2rVrl82jzxYppXT55ZeHsxEjRmRze76wbZp6LzileD+4mveCU6rsfnC0F5xSvB8c7QWnFO8HR/ddKcX3I9W8F/x+s6Krz71xSvH9cTXfG0d7wSnFn4+rYS+4lJZ2f95c99Jamsceeyybf+973ws7o0ePzuZf/epXw06bNm2y+fz588NOdP+QUkrDhw8PZ+WaOnVqOOvTp082L/XfHwYPHlz2Gm6//fZs3qNHj7BT6j4qMmbMmGxe6r6r1H9n+MhHPlL2Gh544IFsfvjhh4edUvsXlVSfz7Rbt25tgJWUp5LPayWf05Ti57U+z2n0OFOKH2tj/f2mVNnHCnwwkydPzuZ77bVX2Dn66KMbaDUNyzcOAwAAAAAAAAAAAEABODgMAAAAAAAAAAAAAAXg4DAAAAAAAAAAAAAAFICDwwAAAAAAAAAAAABQAA4OAwAAAAAAAAAAAEABODgMAAAAAAAAAAAAAAVQ29QLqBbjx4/P5j/72c/Czje/+c1sfvPNN1dkTdASPPXUU+Hs4osvLvvn3Xrrrdm8e/fuYWfw4MHZ/P777w87Z599djh78MEHs/ljjz0Wdo455phsftttt4WdXXfdtaw8pZTOPPPMbH777beHnTFjxmTzKVOmhJ2pU6dm86997WthJ3q+S/1bX3/99eEscscdd4SzT37yk9n82GOPDTs77rhj2WsgpQMPPDCcPfLII9n88ssvDzvdunXL5i+++GLYOeOMM7L5uHHjwk591NbmP061b98+7Gy3XdP/v1vz5s3L5ieccEIjr6QyVq5cmc0nT5
```
Comment (Far0n) [inline h2o-py/demos/xgboost_feature_interactions_demo.ipynb:391]: I think `accuracy` is not super precise in this context. Gain is a rather a measure of loss reduction.

Comment (Far0n) [inline h2o-py/demos/xgboost_feature_interactions_demo.ipynb:393]: `Cover` is a property of a branch or if propagated of a node but not of a feature.

Comment (Far0n) [inline h2o-py/demos/xgboost_feature_interactions_demo.ipynb:395]: `Frequency` is not a measure of gain.

Comment (Far0n) [reply h2o-py/demos/xgboost_feature_interactions_demo.ipynb:395]: Also maybe: "Frequency (FScore)"

**Review by Far0n — CHANGES_REQUESTED** [review]
I think this notebook is well-structured and easy to follow. I just have a few comments regarding the explanations of Gain, Cover and Frequency (Fscore).

**Comment by koniecsveta** [comment]
> I think this notebook is well-structured and easy to follow. I just have a few comments regarding the explanations of Gain, Cover and Frequency (Fscore).

updated as discussed in ce1bdb8

**Review by Far0n — APPROVED** [review]
lgtm now!

**Comment by maurever** [comment]
> @maurever There is a little issue with `plot_tree` on my sys (tested in firefox and chrome with py3.6 and jupyter 1.0.0):
> 
> grafik

Do you think the images are too big? This can be set by this line:

`rcParams['figure.figsize'] = 50, 80`

However, if the size is smaller, then the trees with more layers are too small. There could be some feature to plot image responsibly, but I think for this example it is not important and every user can change the setting of how he/she wants.

**Comment by maurever** [comment]
Thank you, @Far0n; you went through this PR and for your comments!

**Pull request merged**